researcher.lic

```diff
@@ -0,0 +1,178 @@
+      set_researching(false)
+      @current_topic = nil
+    else
+      start_research
+    end
+  end
+
+  def start_research
+    return if researching
+
+    research_start_msgs = [
+      'You focus', 
+      'You tentatively', 
+      'You confidently', 
+      'Abandoning the normal', 
+      'You are already busy',
+      'You start to research',
+      'With a mixture of rational concern',
+      'You expertly coach',
+      'You begin to bend',
+      'You bring yourself',
+      'Inwardly, you make an exercise',
+      'You orient in the figurative direction'
+    ]
+
+    research_start_fail_msgs = [
+      'You cannot begin',
+      'Usage:', 
+      'You do not know how to research'
+    ]
+
+
+    check_gaf
+    @current_topic = 'stream' if @current_topic == 'attunement'
+    case bput("research #{@current_topic} 300", research_start_msgs, research_start_fail_msgs )
+    when 'You cannot begin'
+      DRCA.release_magical_research
+      start_research
+    when *research_start_msgs
+      set_researching(true)
+    when 'Usage:', 'You do not know how to research'
+      DRC.message("Invalid research option")
+      set_researching(false)
+      exit
+    end
+  end
+
+  def check_gaf
+    return if DRSpells.active_spells.include?('Gauge Flow')
+    # Doing the AND check for set and spell is to not rely on lich nil; indexing into a non-existant key raises exception, not nil.
+    # TODO: try something similar to python dictionary .get method instead
+    if @settings['waggle_sets']['gaf'] && @settings['waggle_sets']['gaf']['Gauge Flow']
+      DRCA.cast_spell(@settings['waggle_sets']['gaf']['Gauge Flow'], @settings)
+    else
+      DRC.message("Using min prep. Make a waggle_set called 'gaf' with Gauge Flow in it if you want more control")
+      DRCA.cast_spell({ 'abbrev' => 'gaf', 'mana' => 5}, @settings)
+    end
+  end
+
+  def get_args
+    arg_definitions = [
+      [
+        { name: 'skill',
+          options: %w[
+            attunement augmentation energy field fundamental
+            sorcery spell stream utility warding
+          ], 
+          description: 'What skill or type of research do you want to do?'}
+      ],
+      [
+        { name: 'symbiosis', regex: /symbiosis/i, description: "Do symbiosis research?"},
+        { name: 'sym_type', options: %w[
+          activate avoid cast discern endure
+          examine explore harness harvest
+          heal impress learn perform remember
+          resolve spell spring strengthen watch
+          ], 
+          description: "Which symbiosis do you want to research?"}
+      ],
+      [
+        { name: 'guild', regex: /guild/i, description: "Perform guild specific research"},
+        {
+          name: 'guild_skill', 
+          options: %w[
+            astrology astro plane
+            summoning summon planes
+            theurgy road
+          ],
+          description: "which guild skill?"
+        }
+      ]
+    ]
+    DRC.parse_args(arg_definitions)
```

Comment:
I have a _minor_ preference for `parse_args` to always be the first (or second) thing done in initialize, for the sake of consistency across scripts. If we extracted the arrays into something named, do you think the args would be clear enough that we could move them?